server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -208,6 +208,8 @@ public void validate(Integer numRoutingShards, Map<Setting<Integer>, Integer> se
     public static final String SETTING_VERSION_UPGRADED = "index.version.upgraded";
     public static final String SETTING_VERSION_UPGRADED_STRING = "index.version.upgraded_string";
     public static final String SETTING_CREATION_DATE = "index.creation_date";
+    public static final String SETTING_LIFECYCLE_ORIGINATION_DATE = "index.lifecycle.origination_date";
```

Comment:
This setting should go within the `x-pack` plugin code rather than in the server code, since it's related to the ILM plugin. I would recommend putting it in `LifecycleSettings` since the other ILM-specific settings go there.